metabase/metabase
# Fix TextWidget inconsistent behavior when all chars are removed

Closes https://github.com/metabase/metabase/issues/31606

### Description

When all the characters were removed from the filter, we needed to clearly set value to null as after click on cross icon.

### How to verify

- Go to New Question -> `Select * from products Where category = {{test}}`
- Set test filter to a text variable type
- enter some value, press play
- remove all chars from filter, click outside, press play - you should see an error that filter parameter is not specified (before we used an empty string instead)

### Demo

https://www.loom.com/share/6a25ad4928774f659f618289c0ec139f

### Checklist
